pkg/runtime/wfengine/state/state.go

```diff
@@ -326,22 +329,28 @@ func LoadWorkflowState(ctx context.Context, state state.Interface, actorID strin
 	for i := range metadata.GetInboxLength() {
 		key = getMultiEntryKeyName(inboxKeyPrefix, i)
 		if bulkRes[key] == nil {
-			return nil, fmt.Errorf("failed to load inbox state key '%s': not found", key)
+			// If an inbox entry is missing, it means we caught the state in between saves.
+			// Return the state we have so far. This likely happened due to the reminder firing
+			// before all the inbox entries from the previous transactions went thru in the state store
+			wfLogger.Warnf("Inbox entry '%s' not found, state may be partially saved", key)
+			break
```

Comment:
Why are we doing this? We have to fail here as the workflow is corrupt??

Reply:
I was investigating a race condition with the inbox/metadata not being found and was trying this for a fix. If this is intentional meaning the wf is corrupt I can remove the diff and look elsewhere